code/modules/overmap/contacts/_contacts.dm

```diff
@@ -0,0 +1,103 @@
+#define SENSOR_TIME_DELAY 0.2 SECONDS
+/datum/overmap_contact
+
+	var/name  = "Unknown"            // Contact name.
+	var/image/marker                 // Image overlay attached to the contact.
+	var/pinged = FALSE				 // Used to animate overmap effects.
+	var/list/images = list()		 // Our list of images to cast to users.
+
+	// The sensor console holding this data.
+	var/obj/machinery/computer/ship/sensors/owner
+
+	// The actual overmap effect associated with this.
+	var/obj/effect/overmap/effect
+
+/datum/overmap_contact/New(var/obj/machinery/computer/ship/sensors/creator, var/obj/effect/overmap/source)
+	// Update local tracking information.
+	owner =  creator
+	effect = source
+	name =   effect.name
+	owner.contact_datums[effect] = src
+
+	marker = image(loc = effect, icon = effect.icon, icon_state = effect.icon_state)
```

Comment:
can change `var/image/marker` to `var/mutable_apperance/marker`
and then just do
```
marker = new(effect)
marker.alpha = 0
``` 
here, won't need `name =   effect.name` and the rest of stuff

Reply:
Per Lohikar, it doesn't sound like using a mutable appearance for this is a good idea. I went ahead and copied the appearance of the effect instead and it seems to work just as well while cutting down on a few extraneous lines.